src/BoltOn/Cqrs/CqrsInterceptor.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BoltOn.Logging;
+using BoltOn.Mediator.Interceptors;
+using BoltOn.Mediator.Pipeline;
+
+namespace BoltOn.Cqrs
+{
+	public class CqrsInterceptor : IInterceptor
+	{
+		private readonly EventBag _eventBag;
+		private readonly IBoltOnLogger<CqrsInterceptor> _logger;
+		private readonly IEventDispatcher _eventDispatcher;
+
+		public CqrsInterceptor(EventBag eventBag, IBoltOnLogger<CqrsInterceptor> logger,
+			IEventDispatcher eventDispatcher)
+		{
+			_eventBag = eventBag;
+			_logger = logger;
+			_eventDispatcher = eventDispatcher;
+		}
+
+		public TResponse Run<TRequest, TResponse>(IRequest<TResponse> request,
+			Func<IRequest<TResponse>, TResponse> next) where TRequest : IRequest<TResponse>
+		{
+			var response = next(request);
+			if (_eventBag.EventsToBeProcessed.Any())
+			{
+				throw new NotSupportedException("CQRS not supported for non-async calls");

```

Comment:
@gokulm : why is that?

Reply:
pretty much most the frameworks and libs do not support async, tired of creating a sync and async version. the plan is to get rid of all the sync methods eventually.